nucypher/blockchain/eth/interfaces.py

```diff
@@ -390,3 +394,28 @@ def deploy_contract(self, contract_name: str, *args, **kwargs) -> Tuple[Contract
                                contract_abi=contract_factory.abi)
 
         return contract, txhash
+
+    def call_backend_sign(self, account: str, message: bytes) -> str:
+        """
+        Calls the appropriate signing function for the specified account on the
+        backend. If the backend is based on eth-tester, then it uses the
+        eth-tester signing interface to do so.
+        """
+        provider = self._providers[0]
+        if isinstance(provider, EthereumTesterProvider):
+            address = to_canonical_address(account)
+            sig_key = provider.ethereum_tester.backend._key_lookup[address]
+            signed_message = sig_key.sign_msg(message)
+            return signed_message.to_hex()
+        else:
+            return self.w3.eth.sign(account, data=message) # Technically deprecated...
```

Comment:
...

Reply:
Yeah, this isn't great. Unless we're going with a full wallet implementation before testnet, this is what were stuck with for the current ecosystem.

I don't think handling wallet keys is something we're ready to tackle _right now_.

Reply:
:eyes:

Reply:
> ...

Uff, that's tough...